cubi_tk/sodar/ingest.py

```diff
@@ -0,0 +1,366 @@
+
+    def __init__(self, args):
+        # Command line arguments.
+        self.args = args
+
+        # Path to iRODS environment file
+        self.irods_env_path = Path(Path.home(), ".irods", "irods_environment.json")
+        if not self.irods_env_path.exists():
+            logger.error("iRODS environment file is missing.")
+            sys.exit(1)
+
+        # iRODS environment
+        self.irods_env = None
+
+    def _init_irods(self):
+        """Connect to iRODS."""
+        irods_auth_path = self.irods_env_path.parent.joinpath(".irodsA")
+        if irods_auth_path.exists():
+            try:
+                session = iRODSSession(irods_env_file=self.irods_env_path)
+                session.server_version  # check for outdated .irodsA file
+            except Exception as e:
+                logger.error("iRODS connection failed: %s", self.get_irods_error(e))
+                logger.error("Are you logged in? try 'iinit'")
+                sys.exit(1)
+            finally:
+                session.cleanup()
+        else:
+            # Query user for password.
+            logger.info("iRODS authentication file not found.")
+            password = getpass.getpass(prompt="Please enter SODAR password:")
+            try:
+                session = iRODSSession(irods_env_file=self.irods_env_path, password=password)
+                session.server_version  # check for exceptions
+            except Exception as e:
+                logger.error("iRODS connection failed: %s", self.get_irods_error(e))
+                sys.exit(1)
+            finally:
+                session.cleanup()
+
+        return session
+
+    @classmethod
+    def setup_argparse(cls, parser: argparse.ArgumentParser) -> None:
+        parser.add_argument(
+            "--hidden-cmd", dest="sodar_cmd", default=cls.run, help=argparse.SUPPRESS
+        )
+        group_sodar = parser.add_argument_group("SODAR-related")
+        group_sodar.add_argument(
+            "--sodar-url",
+            default=os.environ.get("SODAR_URL", "https://sodar.bihealth.org/"),
+            help="URL to SODAR, defaults to SODAR_URL environment variable or fallback to https://sodar.bihealth.org/",
+        )
+        group_sodar.add_argument(
+            "--sodar-api-token",
+            default=os.environ.get("SODAR_API_TOKEN", None),
+            help="SODAR API token. Defaults to SODAR_API_TOKEN environment variable.",
+        )
+        parser.add_argument(
+            "-r",
+            "--recursive",
+            default=False,
+            action="store_true",
+            help="Recursively match files in subdirectories. Creates iRODS sub-collections to match directory structure.",
+        )
+        parser.add_argument(
+            "-c",
+            "--checksums",
+            default=False,
+            action="store_true",
+            help="Compute missing md5 checksums for source files.",
+        )
+        parser.add_argument(
```

Comment:
This should definitely be TRUE by default

Reply:
Not sure I agree as this starts writing a bunch of files to your computer. I thought about creating temporary .md5 files and removing them during a cleanup step. Maybe that would be a better option.

Reply:
I think I placed this comment unclearly. I meant the following line here (validating the checksums in irods)

Reply:
I'm a bit hesitant to introduce too much "magic", in this case assuming that the user always wants to compute remote checksums right away. For large files this takes a significant amount of time. There's also an upcoming SODAR feature which would make this flag mostly irrelevant: bihealth/sodar-server/issues/1634